### PR TITLE
Add enum for reuseIdentifier view

### PR DESCRIPTION
**Pull request by phimage**
First I create some object `StoryBoard`, `ViewController`, `Scene`,`Segue`, which extends `XMLObject`, a container for `XMLIndexer`
Then use this objects instead of making call to xml api directly
I think `processStoryboard()` is now more lisible

For #5 enhancement feature
- I do the same as `Segue`, create a `ReusableProtocol`, a `ReusableKind` object, an enum `Reusable` in custom controller (if there is reuseIdentifier children component)
- add some extensions method in controller to use `Reusable` directly

Add some information about Xcode plugin troubleshooting when upgrading xcode

PS: I don't know why the commit "Add Xcode 6.3.2 ID in XNatalie plugin. Fix menu item initialisatio..." is also referenced in this pull request... (not same hash maybe), but github say able to merge...

### CODE REVIEW

**Comment by krzyzanowskim** [comment]
Can you split it in separated PRs?

**Comment by krzyzanowskim** [comment]
registering reusable identifiers looks wired though

File: README.md [hunk 1]
```diff
@@ -76,6 +76,72 @@ extension MyCustomViewController {
...
+Each custom collection and table view controller is extended with this code and provide a list of available reuse identifier
+
+`Reusable` enumeration contains list of available reusable identifier
+
+`kind` property represent types of view (table, collection)
+
+`viewType` property return type of view if specified in storyboard.
+
+```swift
+extension MyCustomTableViewController {
```
Comment (krzyzanowskim) [inline README.md:92]: this is "implementation detail" no need to put it to README. Example of usage is enough.

File: README.md [hunk 2]
```diff
@@ -76,6 +76,72 @@ extension MyCustomViewController {
...
+```swift
+class MyCustomTableViewController: UITableViewController, UITableViewDataSource {
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        return self.dequeueReusableCell(Reusable.myCellId, forIndexPath: indexPath) as! UITableViewCell
+    }
+    )
+```
+
+Before dequeue view you **must** register for each identifier a class or a xib
```
Comment (krzyzanowskim) [inline README.md:126]: it is not required by the Natalie right? it is up to programmer if he register it or not (like for Storyboard cells are automatically registered).

In general, leave example of usage of Natalie. No need to confuse about other things.

**Comment by phimage** [comment]
Ok If you prefer I Will make separated PR next week, one by subject
Ok I Will shorten the readme (I take exemple on segue when doing it)
Yes If programmer want to use deque method , he must register. No things to do for a programmer who don't care about reuseisentifier

**Pull request closed without merging**